### ISSUE TITLE
App freezes consistently

### ISSUE THREAD

**Issue by someoneEsle** [wireapp/wire-ios#1018]
My friend (iPhone 5, iOS 10.3.1) and I (android) have a chat that contains something between 16k and 20k messages. My friend reports that the following happens every time:
- open the conversation with me
- start typing something
- app freezes for several seconds (around 5 or 6)

I would send a log but there is no way to get one that isn't related to voice functionalities only. After sending the first message things get a little better, meaning that the app doesn't freeze consistently on every message my friend is typing, just some of them.

**Comment by someoneEsle** [comment]
Update: we have 18'798 messages in this chat, and chatting has become impossible. When he hits send, sometimes the message gets sent only after a couple of minutes (literally). In the meantime he can receive and send other messages more or less fine, but the fact that some messages take forever to send messes up the whole history of the conversation.
This is another reason why I think there should be an optional automatic trim on all old messages in all conversations. This would also fix some space problems (https://github.com/wireapp/wire-android/issues/318)

**Comment by someoneEsle** [comment]
Update: messages are now taking 4-5 minutes to send very often
This is probably also related to #306 : my friend's app now takes between 10 to 15 seconds just to open

**Comment by typfel** [comment]
Will definitely take look at this when time is available (soon). In theory the amount of messages in conversation should not impact how long it takes to send a new message, something is obviously wrong here.

**Comment by someoneEsle** [comment]
> In theory the amount of messages in conversation should not impact how long it takes to send a new message

True, but my friend doesn't have problems when sending messages in empty conversations, like when he texts Otto. So it must be something related to our chat.

**Comment by someoneEsle** [comment]
Update: today my friend's app has basically stopped working for good. Although he gets notifications, when he opens our chat he doesn't see any new messages.

**Comment by Awerick** [comment]
Just a simple thought I wanted to share:
Until this issue is resolved, maybe it could be a workaround to use a separate group conversation…
I mean, start a group with your friend + a third person (maybe Otto the Bot) and remove the third person.
As this is a separate conversation, this might not be affected by the bug and you could use this group to still use Wire with your contact.

**Comment by highmeh** [comment]
Is there a status update on this? I am having the same issue in one conversation with tens of thousands of messages. Wire support has thus far said to either 1) delete the conversation history (not a good option), or 2) go through every message and manually delete every image, gif, and file. This is not practical as there are many thousands of messages and no way to mass purge. The app is becoming increasingly unusable; every few messages sent, the keyboard locks up and the app freezes entirely. When it becomes usable again (minutes later), the messages often stay in "Sending..." state for several minutes before failing to send. I often have to resend 2-3 times before they go through. This is only an issue on IOS, the desktop app works fine.
